docs/ruleset_functions/is_zone_mechanically_heated_and_not_cooled.md

```diff
@@ -0,0 +1,43 @@
+# is_zone_mechanically_heated_and_not_cooled
+**Schema Version:** 0.0.23  
+
+**Description:** determines whether a zone is mechanically heated, but not cooled.  Checks for transfer air
+
+**Inputs:**
+- **RMI**
+- **zone**
+
+**Returns:**  
+- **result**: a boolean - TRUE or FALSE
+ 
+**Function Call:**
+- **get_list_hvac_systems_associated_with_zone**
+
+## Logic:
+- set a boolean has_heating_system to false: `has_heating_system = FALSE`
+- set a boolean has_cooling_system to false: `has_cooling_system = FALSE`
+- set result to FALSE: `result = FALSE`
+- get the hvac systems serving the zone by using the get_list_hvac_systems_associated_with_zone function: `list_hvac_systems = get_list_hvac_systems_associated_with_zone(RMI,zone)`
+	- loop through each HeatingVentilatingAirconditionsSystem system: `for system in list_hvac_systems:`
+		- check for a heating system: `if system.heating_system != NULL or system.heating_system.type != NONE:`
+			- set the boolean has_heating_system to true: `has_heating_system = TRUE`
+		- look for a cooling system: `if system.cooling_system != NULL or system.cooling_system != NONE:`
+			- there is a cooling system: `has_cooling_system = TRUE`
+- at this point, if there is no cooling system, continue looking in the zone terminals for a cooling system (if there is a cooling system at this level, the function should return false regardless of what is in the zone terminals, and we can skip to the end): `if !has_cooling_system:`
+	- look at each terminal in the zone: `for terminal in zone.terminals:`
+		- check if there is a terminal.heating_source specified: `if terminal.heating_source != NULL or terminal.heating_source != NONE:`
+			- there is a terminal heat source, set has_heating_system equal to true: `has_heating_system = TRUE`
+		- check if there is a terminal.cooling_source specified: `if terminal.cooling_source != NULL or terminal.cooling_source != NONE:`
+			- there is a terminal cooling source, set has_cooling_system equal to true: `has_cooling_system = TRUE`
+- now, if has_heating_system is true and has_cooling_system is false, set result equal to true: `if has_heating_system and !has_cooling_system: result = TRUE`
+- if result is FALSE, check if the zone receives transfer air from a cooled zone: `if !result:`
+	- if the zone has transfer air: `if zone.transfer_airflow_rate > 0:`
+		- set result equal to the result of is_zone_mechanically_heated_and_not_cooled for the transfer air source zone: `result = is_zone_mechanically_heated_and_not_cooled(RMI, zone.transfer_airflow_source_zone)
```

Comment:
I am not sure I understand what line 35 is doing.

Reply:
I don't think this will work because for the zone.transfer_airflow_source_zone we only need to check if that zone itself is mechanically cooled. (It's irrelevant whether it is heated because we only get to this point if the master zone has heating.). 

Also, I don't think we need to account for "cascading" transfer air (when air is transferred to the master zone from a cooled zone via one or more zones that don't have mechanical cooling). The exception is "Exception: Spaces designed with heating only systems ..... not exhausting or transferring air **from mechanically cooled thermal zones** in the proposed design shall not be modeled with mechanical cooling."

Reply:
yes, you are right.  I have updated the function to look expressly at the hvac systems and terminals in the source zone